cypress/pages/NewDataSourcePage.js

```diff
@@ -0,0 +1,31 @@
+import Page from './Page';
+
+class NewDataSourcePage extends Page {
+  constructor() {
+    super('/data_sources/new');
+    this.databaseSourceList = '[data-testid=DatabaseSourceList]';
+    this.dynamicForm = '[data-testid=DynamicForm]';
+    this.saveButton = '[data-testid=SaveButton]';
+  }
+
+  selectSource(sourceName) {
+    cy.get(this.databaseSourceList)
+      .contains(sourceName)
+      .click();
+  }
+
+  fill(fieldName, value) {
+    cy.get(this.dynamicForm)
+      .get('[data-testid="' + fieldName + '"]')
+      .clear()
+      .type(value);
+
+    return this;
```

Comment:
You could also put the submit button in ```fill()``` as these actions are usually done at the same time.